Modules/BarcodeScanner/BarcodeScannerView.swift

```diff
@@ -0,0 +1,320 @@
+        captureSessionQueue.async { [weak self] in
+            guard let self = self else { return }
+
+            self.captureSession.beginConfiguration()
+            defer { self.captureSession.commitConfiguration() }
+
+            if self.captureSession.canSetSessionPreset(.high) {
+                self.captureSession.sessionPreset = .high
+            }
+
+            guard let captureDevice = self.captureDevice else {
+                return DispatchQueue.main.async { [weak self] in
+                    self?.model.onError(.videoCapturingDeviceNotAvailable)
+                }
+            }
+
+            guard let captureDeviceInput = try? AVCaptureDeviceInput(device: captureDevice) else {
+                return DispatchQueue.main.async { [weak self] in
+                    self?.model.onError(.videoCapturingConfigurationFailed)
+                }
+            }
+
+            guard self.captureSession.canAddInput(captureDeviceInput) else { return }
+
+            self.captureSession.addInput(captureDeviceInput)
+            self.captureDeviceInput = captureDeviceInput
+
+            guard self.captureSession.canAddOutput(self.captureMetadataOutput) else {
+                return DispatchQueue.main.async { [weak self] in
+                    self?.model.onError(.videoCapturingConfigurationFailed)
+                }
+            }
+
+            self.captureSession.addOutput(self.captureMetadataOutput)
+            self.captureMetadataOutput.metadataObjectTypes = self.model.barcodeTypes.map { $0.rawValue }
+            self.captureMetadataOutput.setMetadataObjectsDelegate(self, queue: .main)
+            self.isCaptureSessionConfigured = true
+
+            DispatchQueue.main.async { [weak self] in
+                self?.setVideoOrientation(to: UIApplication.shared.statusBarOrientation)
+                self?.didChangeState()
+            }
+        }
+    }
+
+    public override func didChangeModel() {
+        super.didChangeModel()
+
+        captureSessionQueue.async { [weak self] in
+            guard let self = self, self.isCaptureSessionConfigured else { return }
+
+            self.captureSession.beginConfiguration()
+            defer { self.captureSession.commitConfiguration() }
+
+            self.captureMetadataOutput.metadataObjectTypes = self.model.barcodeTypes.map { $0.rawValue }
+        }
+    }
+
+    public override func willMove(toSuperview newSuperview: UIView?) {
+        switch (superview, newSuperview) {
+        case (.none, .some):
+            isScanningEnabled = true
+
+        case (.some, .none):
+            isScanningEnabled = false
+
+        default:
+            break
+        }
+    }
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+
+        captureVideoPreviewLayer.frame = bounds
+    }
+
+    private func didChangeState() {
+        if isScanning, isScanningEnabled, !isCaptureSessionRunning {
+            self.captureSessionQueue.async { [weak self] in
```

Comment:
There is already a startScanning method which is public we could add another one too but this is the only location where the call happens so I prefer inlining the code here for better readability instead of jumping through methods while reading the code ;)